main.js

```diff
@@ -77,28 +79,28 @@ function deleteIdeaCard(ideaId) {
       var ideaToDelete = ideas[i];
       ideaToDelete.deleteFromStorage();
       ideas.splice(i, 1);
-      displayCards();
     }
   }
 }
 
-function favoriteIdeaCard() {
-
+function favoriteIdeaCard(ideaId) {
+  for (var i = 0; i < ideas.length; i++) {
+    if (ideaId === ideas[i].id) {
+      ideas[i].star = !ideas[i].star;
+      ideas[i].updateIdea();
+    }
+  }
 }
 
 function handleIdeaCardGridClick(event) {
+  var ideaId = Number(event.target.closest('.box-container').id);
```

Comment:
I pulled this out of the if-statements because the ideaId will be used in both of the if-statements.